### PR TITLE
Store model parsing logs in SimInt API

### PR DESCRIPTION
**Pull request by lpereiracgn**

### CODE REVIEW

File: Cognite.Simulator.Utils/StagingArea.cs [hunk 1]
```diff
@@ -19,161 +8,7 @@ namespace Cognite.Simulator.Utils
...
-                return default;
-            }
-            catch (Exception e)
-            {
-                _logger.LogError("Could not fetch data from CDF RAW", e);
-                return default;
-            }
-        }
-    }
+    public class StagingArea {}
```
Comment (polomani) [inline Cognite.Simulator.Utils/StagingArea.cs:11]: why do we need this empty class?

File: Cognite.Simulator.Tests/UtilsTests/FileLibraryTest.cs [hunk 2]
```diff
@@ -92,17 +92,24 @@ await modelLibTasks
...
-                Assert.Equal(ParsingStatus.success, log2.Status);
+                var log1 = await cdf.Alpha.Simulators.RetrieveSimulatorLogsAsync(
+                    new List<Identity> { new Identity(v1.LogId) }, source.Token).ConfigureAwait(false);
+
+                var log1Data = log1.First().Data;
+                Assert.NotEmpty(log1Data);
+
+                var parsedModelEntry1 = log1Data.Where(lg => lg.Message.StartsWith("Model parsed successfully"));
+                Assert.Equal("Model parsed successfully", parsedModelEntry1.First().Message);
+                Assert.Equal("Information", parsedModelEntry1.First().Severity);
```
Comment (polomani) [inline Cognite.Simulator.Tests/UtilsTests/FileLibraryTest.cs:103]: is it essencial to check it twice for each or are we really only interested in the latest?

File: Cognite.Simulator.Tests/UtilsTests/FileLibraryTest.cs [hunk 3]
```diff
@@ -373,6 +378,7 @@ protected override TestFileState StateFromModelRevision(SimulatorModelRevision m
                 Processed = false,
                 Version = modelRevision.VersionNumber,
                 ExternalId = modelRevision.ExternalId,
+                LogId = modelRevision.LogId ?? 0,
```
Comment (polomani) [inline Cognite.Simulator.Tests/UtilsTests/FileLibraryTest.cs:381]: we can make LogId optional in TestFileState so we don't have to do this `?? 0`

File: Cognite.Simulator.Utils/ModelLibraryBase.cs [hunk 4]
```diff
@@ -197,14 +197,23 @@ protected virtual async Task ExtractModelInformationAndUpdateState(CancellationT
...
-                finally
-                {
-                    await UpdateModelParsingInfo(group, token).ConfigureAwait(false);
+                foreach (var item in group){
+                    var logId = item.LogId;
+                    using (LogContext.PushProperty("LogId", logId)) {
+                        try
+                        {
+                            Console.WriteLine("PUSHING LOGID");
+                            Console.WriteLine(logId);
```
Comment (polomani) [inline Cognite.Simulator.Utils/ModelLibraryBase.cs:206]:
```suggestion
```

File: Cognite.Simulator.Utils/ModelLibraryBase.cs [hunk 5]
```diff
@@ -197,14 +197,23 @@ protected virtual async Task ExtractModelInformationAndUpdateState(CancellationT
...
+                            _logger.LogInformation("Extracting model information for {ModelName} v{Version}", item.ModelName, item.Version);
+                            await ExtractModelInformation(item, token).ConfigureAwait(false);
+                        }
+                        finally
+                        {
+                            await UpdateModelParsingInfo(item, token).ConfigureAwait(false);
+                        }
+                    } 
                 }
+                await _remoteApiSink.Flush(CdfSimulatorResources, CancellationToken.None).ConfigureAwait(false);
```
Comment (polomani) [inline Cognite.Simulator.Utils/ModelLibraryBase.cs:216]: we decided to flush the logs on the heartbeat. no need to do it here

File: Cognite.Simulator.Utils/ModelLibraryBase.cs [hunk 6]
```diff
@@ -28,10 +29,8 @@ public abstract class ModelLibraryBase<T, U, V> : FileLibrary<T, U>, IModelProvi
         where U : ModelStateBasePoco
         where V : ModelParsingInfo, new()
     {
-        /// <summary>
-        /// Staging area, used to store the model parsing info in CDF
-        /// </summary>
-        protected StagingArea<V> Staging { get; }
+        private readonly ILogger _logger;
+        private readonly ScopedRemoteApiSink _remoteApiSink;
```
Comment (polomani) [inline Cognite.Simulator.Utils/ModelLibraryBase.cs:33]:
```suggestion
```

we decided to flush the logs on the heartbeat. no need to have the ScopedRemoteApiSink here

File: Cognite.Simulator.Utils/ModelLibraryBase.cs [hunk 7]
```diff
@@ -216,55 +225,59 @@ protected virtual async Task ExtractModelInformationAndUpdateState(CancellationT
...
         }
 
-        private async Task UpdateModelParsingInfo(IEnumerable<T> modelStates, CancellationToken token)
+        private async Task UpdateModelParsingInfo(T modelState, CancellationToken token)
         {
-            foreach(var revision in modelStates)
+            if (modelState.ParsingInfo != null && modelState.ParsingInfo.Status != ParsingStatus.ready)
             {
-                if (revision.ParsingInfo != null && revision.ParsingInfo.Status != ParsingStatus.ready)
+                //await Staging.UpdateEntry(modelState.Id, (V) modelState.ParsingInfo, token).ConfigureAwait(false);
```
Comment (polomani) [inline Cognite.Simulator.Utils/ModelLibraryBase.cs:232]:
```suggestion
```

**Review by abdullah-cognite — APPROVED** [review]

**Pull request merged**